server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/HealthCheckRoutes.java

```diff
@@ -65,27 +74,49 @@ public String getBasePath() {
 
     @Override
     public void define(Service service) {
-        this.service = service;
-
-        validateHealthchecks();
+        service.get(HEALTHCHECK, this::validateHealthchecks, jsonTransformer);
+        service.get(HEALTHCHECK + "/checks/:componentName", this::performHealthCheckForComponent, jsonTransformer);
     }
 
     @GET
     @ApiOperation(value = "Validate all health checks")
     @ApiResponses(value = {
-            @ApiResponse(code = HttpStatus.OK_200, message = "OK"),
-            @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
-                message = "Internal server error - When one check has failed.")
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+            message = "Internal server error - When one check has failed.")
+    })
+    public Object validateHealthchecks(Request request, Response response) {
+        List<Result> anyUnhealthyOrDegraded = retrieveUnhealthyOrDegradedHealthChecks();
+
+        anyUnhealthyOrDegraded.forEach(this::logFailedCheck);
+        response.status(getCorrespondingStatusCode(anyUnhealthyOrDegraded));
+        return response;
+    }
+    
+    @GET
+    @Path("/checks/{componentName}")
+    @ApiOperation(value = "Perform the component's health check")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            name = "componentName",
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            defaultValue = "None",
+            example = "/checks/Cassandra%20Backend",
+            value = "The URL encoded name of the component to check.")
     })
-    public void validateHealthchecks() {
-        service.get(HEALTHCHECK,
-            (request, response) -> {
-                List<Result> anyUnhealthyOrDegraded = retrieveUnhealthyOrDegradedHealthChecks();
-
-                anyUnhealthyOrDegraded.forEach(this::logFailedCheck);
-                response.status(getCorrespondingStatusCode(anyUnhealthyOrDegraded));
-                return response;
-            });
+    public Object performHealthCheckForComponent(Request request, Response response) {
+        String componentName = request.params("componentName");
+        Optional<HealthCheck> optHealthCheck = healthChecks.stream()
+            .filter(c -> c.componentName().getName().equals(componentName))
+            .findFirst();
```

Comment:
could use `orElseThrow()` right here like you did with `ResultStatus`

Reply:
I'm not sure if I'm getting you right. But instead of extracting to a local `Optional` and then using `orElseThrow()` on that `Optional` in the next statement  I will join the operations into one statement `findFirst().orElseThrow()`.

Reply:
fixed in 8aae6496c4efb536c4e4c5f740f7401bccb0a101